controllers/clusters/kafka_controller.go

```diff
@@ -234,9 +236,183 @@ func (r *KafkaReconciler) handleUpdateCluster(
+		return err
+	}
+
+	if _, exist := u.Status.ClustersEvents[kafka.Status.ID]; exist {
+		l.Info("User is already existing on the cluster",
+			"user reference", userRef)
+		r.EventRecorder.Eventf(kafka, models.Normal, models.CreationFailed,
+			"User is already existing on the cluster. User reference: %v", userRef)
+
+		return nil
+	}
+
+	patch := u.NewPatch()
+
+	if u.Status.ClustersEvents == nil {
+		u.Status.ClustersEvents = make(map[string]string)
+	}
+
+	u.Status.ClustersEvents[kafka.Status.ID] = models.CreatingEvent
+
+	err = r.Status().Patch(ctx, u, patch)
+	if err != nil {
+		l.Error(err, "Cannot patch the Kafka User status with the CreatingEvent",
+			"cluster name", kafka.Spec.Name, "cluster ID", kafka.Status.ID)
+		r.EventRecorder.Eventf(kafka, models.Warning, models.CreationFailed,
+			"Cannot add Kafka User to the cluster. Reason: %v", err)
+
+		return err
+	}
+
+	return nil
+}
+
+func (r *KafkaReconciler) handleDeleteUser(
+	ctx context.Context,
+	kafka *v1beta1.Kafka,
+	l logr.Logger,
+	userRef *v1beta1.UserReference,
+) error {
+	req := types.NamespacedName{
+		Namespace: userRef.Namespace,
+		Name:      userRef.Name,
+	}
+
+	u := &kafkamanagementv1beta1.KafkaUser{}
+
+	err := r.Get(ctx, req, u)
+
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			l.Info("kafka user is not found", "request", req)
+			r.EventRecorder.Event(kafka, models.Warning, "Not Found",
+				"User is not found, create a new one Kafka User or provide correct userRef.")
+
+			return err
+		}
+
+		l.Error(err, "Cannot get Kafka user", "user", u.Spec)
+		r.EventRecorder.Eventf(kafka, models.Warning, models.DeletionFailed,
+			"Cannot get Kafka user. user reference: %v", userRef)
+		return err
+	}
+
+	if _, exist := u.Status.ClustersEvents[kafka.Status.ID]; !exist {
+		l.Info("User is not existing on the cluster",
+			"user reference", userRef)
+		r.EventRecorder.Eventf(kafka, models.Normal, models.DeletionFailed,
+			"User is not existing on the cluster. User reference: %v", userRef)
+
+		return nil
+	}
+
+	patch := u.NewPatch()
+
+	u.Status.ClustersEvents[kafka.Status.ID] = models.DeletingEvent
+
+	err = r.Status().Patch(ctx, u, patch)
+	if err != nil {
+		l.Error(err, "Cannot patch the Kafka User status with the DeletingEvent",
+			"cluster name", kafka.Spec.Name, "cluster ID", kafka.Status.ID)
+		r.EventRecorder.Eventf(kafka, models.Warning, models.DeletionFailed,
+			"Cannot patch the Kafka User status with the DeletingEvent. Reason: %v", err)
+
+		return err
+	}
+
+	return nil
+}
+
+func (r *KafkaReconciler) handleUserEvent(
+	newObj *v1beta1.Kafka,
+	oldUsers []*v1beta1.UserReference,
+) {
+	ctx := context.TODO()
+	l := log.FromContext(ctx)
+
+	for _, newUser := range newObj.Spec.UserRefs {
+		var exist bool
+
+		for _, oldUser := range oldUsers {
+			if *newUser == *oldUser {
+				exist = true
+				break
+			}
+		}
+		if exist {
+			continue
+		}
+		err := r.handleCreateUser(ctx, newObj, l, newUser)
```

Comment:
Please fix similar cases
```suggestion
		}
		
		if exist {
			continue
		}
		
		err := r.handleCreateUser(ctx, newObj, l, newUser)
```

Reply:
done